matchzoo/models/anmm_model.py

```diff
@@ -0,0 +1,68 @@
+"""An implementation of aNMM Model."""
+
+
+import logging
+import keras
+
+from keras.activations import softmax
+from keras.initializers import RandomUniform
+
+from matchzoo import engine
+
+logger = logging.getLogger(__name__)
+
+
+class ANMMModel(engine.BaseModel):
+    """
+    ANMM Model.
+
+    Examples:
+        >>> model = ANMMModel()
+        >>> model.guess_and_fill_missing_params(verbose=0)
+        >>> model.build()
+
+    """
+
+    @classmethod
+    def get_default_params(cls) -> engine.ParamTable:
+        """:return: model default parameters."""
+        params = super().get_default_params(with_embedding=True)
+        params.add(engine.Param('bin_num', 60))
+        params.add(engine.Param('dropout_rate', 0.1))
+        params.add(engine.Param('left_text_len', 10))
+        params.add(engine.Param('num_layers', 2))
+        params.add(engine.Param('hidden_sizes', [10, 10]))
+        return params
+
+    def build(self):
+        """
+        Build model structure.
+
+        aNMM model based on bin weighting and query term attentions
+        """
+        # query is [batch_size, left_text_len]
+        # doc is [batch_size, right_text_len, bin_num]
+        query, doc = self._make_inputs()
+        embedding = self._make_embedding_layer()
+        initializer_fc = RandomUniform(minval=-0.1, maxval=0.1, seed=11)
+        initializer_gate = RandomUniform(minval=-0.01, maxval=0.01, seed=11)
+
+        q_embed = embedding(query)
+        q_w = keras.layers.Dense(
+            1, kernel_initializer=initializer_gate, use_bias=False)(q_embed)
+        q_w = keras.layers.Lambda(
```

Comment:
For the attention layer, you can refer to [drmm](https://github.com/NTMC-Community/MatchZoo/blob/feature/add_drmm/matchzoo/models/drmm.py) to implement the same function. We will extract these functions out to form the common layer.

Reply:
I will fix this after the attention function has been included into the common layer